### PR TITLE
(test) Adding test coverage for the encounter provider handler

### PR DESCRIPTION
**Pull request by gitcliff**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary
Adding test coverage for the encounter provider handler for handlingFieldSubmission, getting initial and previous values

### CODE REVIEW

File: src/submission-handlers/encounterProviderHandler.test.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    };
+    // replay
+    const provider = EncounterProviderHandler.handleFieldSubmission(field, 'New Provider', encounterContext);
+    // verify
+    expect(provider).toEqual('New Provider');
+  });
+});
+
+describe('EncounterProviderHandler - getInitialValue', () => {
+  // new submission (enter mode)
```
Comment (samuelmale) [inline src/submission-handlers/encounterProviderHandler.test.ts:90]: Is this testing submission logic?

File: src/submission-handlers/encounterProviderHandler.test.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      rendering: 'number',
+    },
+    validators: [],
+    meta: {},
+  },
+];
+
+describe('EncounterProviderHandler - handleFieldSubmission', () => {
+  // new submission (enter mode)
+  it('should handle encounter provider submission for ui-select-extended input', () => {
```
Comment (samuelmale) [inline src/submission-handlers/encounterProviderHandler.test.ts:70]: Can you exclude the rendering "ui-select-extended"?

File: src/submission-handlers/encounterProviderHandler.test.ts [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+      allFormFieldsMock,
+      encounterContext,
+    );
+    // verify
+    expect(provider).toEqual('2c95f6f5-788e-4e73-9079-5626911231fa');
+  });
+});
+
+describe('EncounterProviderHandler - getPreviousValue', () => {
+  it('should get previous value for ui-select-extended rendering', () => {
```
Comment (samuelmale) [inline src/submission-handlers/encounterProviderHandler.test.ts:151]: Same here, remove the rendering

**Review by samuelmale — COMMENTED** [review]
Can you move the tests to a single suite?

File: src/submission-handlers/encounterProviderHandler.test.ts (hunk 1 above)
Comment (gitcliff) [reply src/submission-handlers/encounterProviderHandler.test.ts:90]: oh thanks ,,,going to remove it

File: src/submission-handlers/encounterProviderHandler.test.ts [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+  encounterDate: new Date(2020, 11, 29),
+  setEncounterDate: (value) => {},
+  encounterProvider: '2c95f6f5-788e-4e73-9079-5626911231fa',
+  setEncounterProvider: jest.fn,
+  setEncounterLocation: jest.fn,
+  encounterRole: '6d95f6f5-788e-4e73-9079-5626911231f4',
+  setEncounterRole: jest.fn,
+};
+
+const encounterContext2: EncounterContext = {
```
[the remark this answers is not in the document]
Comment (gitcliff) [reply src/submission-handlers/encounterProviderHandler.test.ts:27]: yes i needed one where the encounter is undefined to cater for https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/submission-handlers/encounterProviderHandler.ts#L17

[the remark this answers is not in the document]
Comment (samuelmale) [reply src/submission-handlers/encounterProviderHandler.test.ts:27]: What stops you from:

```ts
const provider = EncounterProviderHandler.getInitialValue(
      null,
      field,
      allFormFieldsMock,
      encounterContext,
    );

```

**Comment by gitcliff** [comment]
thanks @samuelmale ..
i have updated

**Review by samuelmale — APPROVED** [review]
LGTM, thanks @gitcliff!

**Pull request merged**